### PR TITLE
MINIFICPP-1345 Add flake8 check for python files

### PR DESCRIPTION
**Pull request by lordgamez**
Thank you for submitting a contribution to Apache NiFi - MiNiFi C++.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced
     in the commit message?

- [ ] Does your PR title start with MINIFICPP-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically main)?

- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE file?
- [ ] If applicable, have you updated the NOTICE file?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions CI results for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: run_flake8.sh [hunk 1]
```diff
@@ -0,0 +1,6 @@
+#!/bin/bash
+
+set -euo pipefail
+
+directory=${1:-.}
+flake8 --exclude thirdparty,build --ignore E501,W504 --per-file-ignores="steps.py:F811" "${directory}"
```
Comment (szaszm) [inline run_flake8.sh:6]: I'm happy with ignoring [E501](https://www.flake8rules.com/rules/E501.html), but asking for an explanation of ignoring [W504](https://www.flake8rules.com/rules/W504.html) and [F811](https://www.flake8rules.com/rules/F811.html).

Comment (lordgamez) [reply run_flake8.sh:6]: Flake8 has both [W503](https://www.flake8rules.com/rules/W503.html) and [W504](https://www.flake8rules.com/rules/W504.html) enabled by default and they contradict each other. We have to choose one of them to use, in this case I chose to use W503 and ignore W504 to have the binary operators at the end of the line.

[F811](https://www.flake8rules.com/rules/F811.html) is only ignored in steps.py as it defines the `behave` framework steps. This file has the same `step_impl` function name for each step and `flake8` detects it as a redefinition.

Comment (szaszm) [reply run_flake8.sh:6]: Between W503 and W504, I would go for the other one based on this: https://www.python.org/dev/peps/pep-0008/#should-a-line-break-before-or-after-a-binary-operator
> In Python code, it is permissible to break before or after a binary operator, as long as the convention is consistent locally. For new code Knuth's style is suggested.

Knuth's style is break before the operator.

Comment (lordgamez) [reply run_flake8.sh]: Yes, in hindsight it makes more sense to go with that, even the [flake8 rule](https://www.flake8rules.com/rules/W503.html) mentions that this goes against the pep8 recommendation. I added the change in f417b932e4f76b74351c78739eefd1d462dac4b4

File: docker/test/integration/minifi/processors/DeleteS3Object.py [hunk 2]
```diff
@@ -1,13 +1,16 @@
...
 class DeleteS3Object(Processor):
-    def __init__(self,
-        proxy_host = '',
-        proxy_port = '',
-        proxy_username = '',
-        proxy_password = ''):
-            super(DeleteS3Object, self).__init__('DeleteS3Object',
-            properties = {
+    def __init__(
+            self,
```
Comment (szaszm) [inline docker/test/integration/minifi/processors/DeleteS3Object.py:6]: In most cases, continuation indentation was changed to one level of indentation, but here it's changed to two. Why?

Comment (szaszm) [reply docker/test/integration/minifi/processors/DeleteS3Object.py:6]: Not sure about python guidelines in this regard, but my recommendation would be to stick to two levels of indentation for continuations.

Comment (szaszm) [reply docker/test/integration/minifi/processors/DeleteS3Object.py:6]: why not one level: to clearly separate it from normal code one layer deeper
why not aligned: to support find/replace and rename refactoring without messing up the layout

Comment (lordgamez) [reply docker/test/integration/minifi/processors/DeleteS3Object.py:6]: In that case we would have 2 problems with flake8:
./docker/test/integration/minifi/processors/DeleteS3Object.py:10:5: E125 continuation line with same indent as next logical line
./docker/test/integration/minifi/processors/DeleteS3Object.py:11:13: E117 over-indented

Comment (lordgamez) [reply docker/test/integration/minifi/processors/DeleteS3Object.py:6]: The [pep8 recommendation](https://www.python.org/dev/peps/pep-0008/#indentation) is the same in this case:
```
# Add 4 spaces (an extra level of indentation) to distinguish arguments from the rest.
def long_function_name(
        var_one, var_two, var_three,
        var_four):
    print(var_one)
```

Comment (szaszm) [reply docker/test/integration/minifi/processors/DeleteS3Object.py:6]: Let's go with pep8 in this case

File: run_flake8.sh [hunk 3]
```diff
@@ -0,0 +1,6 @@
+#!/bin/bash
+
+set -euo pipefail
+
+directory=${1:-.}
+flake8 --exclude thirdparty,build --ignore E501,W503 --per-file-ignores="steps.py:F811" "${directory}"
```
Comment (fgerlits) [inline run_flake8.sh:6]: If we add `--builtins log,REL_SUCCESS,REL_FAILURE` here, then we can get rid of most of the `# noqa: F821` comments.

Comment (lordgamez) [reply run_flake8.sh]: Good idea! Updated in 51b7f780e0497c5695321631e9c7bd0f655ddcc1

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -156,10 +156,10 @@ jobs:
       - id: install_deps
         run: |
           sudo apt update
-          sudo apt install -y ccache libfl-dev libpcap-dev libboost-all-dev
+          sudo apt install -y ccache libfl-dev libpcap-dev libboost-all-dev flake8
           echo "PATH=/usr/lib/ccache:$PATH" >> $GITHUB_ENV
       - id: build
-        run: ./bootstrap.sh -e -t && cd build  && cmake -DUSE_SHARED_LIBS= -DCMAKE_BUILD_TYPE=Release -DENABLE_BUSTACHE=ON -DENABLE_SQLITE=ON -DENABLE_PCAP=ON -DSTRICT_GSL_CHECKS=AUDIT -DFAIL_ON_WARNINGS=ON .. && make -j4 VERBOSE=1  && make test ARGS="--timeout 300 -j2 --output-on-failure"
+        run: ./bootstrap.sh -e -t && cd build  && cmake -DUSE_SHARED_LIBS= -DCMAKE_BUILD_TYPE=Release -DENABLE_BUSTACHE=ON -DENABLE_SQLITE=ON -DENABLE_PCAP=ON -DSTRICT_GSL_CHECKS=AUDIT -DFAIL_ON_WARNINGS=ON .. && make -j4 VERBOSE=1  && make test ARGS="--timeout 300 -j2 --output-on-failure" && make flake8
```
[the remark this answers is not in the document]
Comment (lordgamez) [reply .github/workflows/ci.yml]: I agree, updated in 51b7f780e0497c5695321631e9c7bd0f655ddcc1

**Review by fgerlits — APPROVED** [review]

**Review by szaszm — APPROVED** [review]

**Pull request closed without merging**